File: WOT-iOS/WOT-iOS/core/metrics/WOTMetric.m
//
//  WOTMetric.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 7/23/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTMetric.h"


@interface WOTMetric ()

@property (nonatomic, readwrite, strong)NSString *metricName;
@property (nonatomic, readwrite, strong)WOTTankMetricEvaluator evaluator;

@end


@implementation WOTMetric

- (id)initWithMetricName:(NSString *)ametricName evaluator:(WOTTankMetricEvaluator)aevaluator {
    
    self = [super init];
    if (self){
        
        self.metricName = ametricName;
        self.evaluator = aevaluator;
    }
    return self;
}

@end

File: WOT-iOS/WOTApi/source/pivot/metrics/WOTMetric.swift
//
//  WOTMetric.swift
//  WOTPivot
//
//  Created by Pavel Yeshchyk on 3/2/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

// MARK: - WOTTankMetricEvaluatorProtocol

@objc
public protocol WOTTankMetricEvaluatorProtocol: NSObjectProtocol {
    func evaluate(list: WOTTanksIDListProtocol) -> WOTTankEvaluationResultProtocol?
}

// MARK: - WOTMetricProtocol

@objc
public protocol WOTMetricProtocol: NSObjectProtocol {
    var metricName: String? { get set }
    var evaluator: WOTTankMetricEvaluatorProtocol? { get set }
    var grouppingName: String? { get set }
}

// MARK: - WOTMetric

@objc public class WOTMetric: NSObject, WOTMetricProtocol {

    @objc
    public required init(metricName: String?, grouppingName: String?, evaluator: WOTTankMetricEvaluatorProtocol?) {
        self.metricName = metricName
        self.evaluator = evaluator
        self.grouppingName = grouppingName
    }

    public var metricName: String?
    public var evaluator: WOTTankMetricEvaluatorProtocol?
    public var grouppingName: String?
}
